lib/service/service_test.go

```diff
@@ -1842,6 +1847,9 @@ func TestInitDatabaseService(t *testing.T) {
 			exitPayload, ok := event.Payload.(ExitEventPayload)
 			require.True(t, ok, "expected ExitEventPayload but got %T", event.Payload)
 			require.Equal(t, "db.init", exitPayload.Service.Name())
+			// Database service init is a critical service, meaning failures on
+			// it should cause the process to exit with error.
+			require.Error(t, eg.Wait())
```

Comment:
If the requires above fails what will make `process` stop? Would this line block up to 20s in that case?

(If you take my comment about the t.Cleanup I think this gets solved too.)